server/src/test/java/org/elasticsearch/index/mapper/vectors/DenseVectorFieldMapperTests.java

```diff
@@ -569,10 +578,83 @@ public void testMaxDimsByteVector() throws IOException {
         assertArrayEquals("Parsed vector is not equal to original.", vector, vectorField.vectorValue());
     }
 
+    @Override
+    protected void assertFetchMany(MapperService mapperService, String field, Object value, String format, int count) throws IOException {
+        assumeFalse("Dense vectors currently don't support multiple values in the same field", false);
+    }
+
+    /**
+     * Dense vectors don't support doc values or string representation (for doc value parser/fetching).
+     * We may eventually support that, but until then, we only verify that the parsing and fields fetching matches the provided value object
+     */
+    @Override
+    protected void assertFetch(MapperService mapperService, String field, Object value, String format) throws IOException {
+        MappedFieldType ft = mapperService.fieldType(field);
+        MappedFieldType.FielddataOperation fdt = MappedFieldType.FielddataOperation.SEARCH;
+        SourceToParse source = source(b -> b.field(ft.name(), value));
+        SearchExecutionContext searchExecutionContext = mock(SearchExecutionContext.class);
+        when(searchExecutionContext.isSourceEnabled()).thenReturn(true);
+        when(searchExecutionContext.sourcePath(field)).thenReturn(Set.of(field));
+        when(searchExecutionContext.getForField(ft, fdt)).thenAnswer(inv -> fieldDataLookup(mapperService).apply(ft, () -> {
+            throw new UnsupportedOperationException();
+        }, fdt));
+        ValueFetcher nativeFetcher = ft.valueFetcher(searchExecutionContext, format);
+        ParsedDocument doc = mapperService.documentMapper().parse(source);
+        withLuceneIndex(mapperService, iw -> iw.addDocuments(doc.docs()), ir -> {
+            Source s = SourceProvider.fromStoredFields().getSource(ir.leaves().get(0), 0);
+            nativeFetcher.setNextReader(ir.leaves().get(0));
+            List<Object> fromNative = nativeFetcher.fetchValues(s, 0, new ArrayList<>());
+            DenseVectorFieldType denseVectorFieldType = (DenseVectorFieldType) ft;
+            switch (denseVectorFieldType.getElementType()) {
+                case BYTE -> {
+                    assumeFalse("byte element type testing not currently added", false);
+                }
+                case FLOAT -> {
+                    float[] fetchedFloats = new float[denseVectorFieldType.getVectorDimensions()];
+                    int i = 0;
+                    for (var f : fromNative) {
+                        assert f instanceof Number;
+                        fetchedFloats[i++] = ((Number)f).floatValue();
+                    }
+                    assertThat("fetching " + value, fetchedFloats, equalTo(value));
+                }
+            }
+        });
+    }
+
+    @Override
+    // TODO: add `byte` element_type tests
```

Comment:
I kept running into an error where the `byte[]` was stored and read as a binary string value. Somewhere along the test path, something is reading the `byte[]` as a string and breaking the test. I could not figure out where, so I simply pushed the `float` changes :)